plugins/quote.py

```diff
@@ -146,6 +146,61 @@ def get_quote_by_chan(db, chan, num=False):
     return format_quote(data, num, count)
 
 
+find_re = re.compile(r"(?:(#\S+)\s+)?([^#]\S+)\s+(.*)$")
+@hook.command('findquote')
+def find_quote(event, db, text):
+    """[#chan] [nick] <to match> OR [nick] <to match> - get the first quote matching <to match> by [nick] in [#chan]"""
+
+    errstr = "No quotes found"
+    find = find_re.match(text)
+    if not find:
+        event.notice_doc()
+        return
+
+    chan, nick, to_match = find.groups()
+
+    query = select([qtable.c.time, qtable.c.nick, qtable.c.msg]) \
+        .where(qtable.c.deleted != 1) \
+        .order_by(qtable.c.time)
+
+    if nick:
+        errstr += " for {}".format(nick)
+        query = query.where(qtable.c.nick == nick)
+    if chan:
+        errstr += " in {}".format(chan)
+        query = query.where(qtable.c.chan == chan)
+
+    allquotes = db.execute(query.order_by(qtable.c.time)).fetchall()
```

Comment:
It would be more efficient to just grab all quotes in the first place and filter them python side, instead of doing multiple DB queries.